gnsi/acltrie/trie.go

```diff
@@ -39,7 +42,29 @@ type trieNode struct {
 	parent    *trieNode
 }
 
-// policies are map of mode to user to action.
+// getAction returns the action associated for the given user and mode.
+// It returns UNSPECIFIED, if the user does not have an action in the node.
+// It returns true, if the policy was for the user, and false if it was for the group.
+func (tn *trieNode) getAction(user string, mode pathzpb.Mode, memberships map[string]map[string]bool) (pathzpb.Action, bool) {
+	if m, ok := tn.users[mode]; ok {
+		if act, ok := m[user]; ok {
+			return act, true
+		}
+	}
+
+	var act pathzpb.Action
+	for group, action := range tn.groups[mode] {
+		if _, ok := memberships[group][user]; ok {
+			if action == pathzpb.Action_ACTION_DENY { // DENY action take precedence over PERMIT.
+				return action, false
+			}
+			act = action
```

Comment:
technically this should be a switch case with a default statement returning error in case we add another enum value.

Reply:
added check on insert